packages/stargate/src/signingstargateclient.ts

```diff
@@ -397,7 +397,12 @@ export class SigningStargateClient extends StargateClient {
     if (!accountFromSigner) {
       throw new Error("Failed to retrieve account from signer");
     }
-    const pubkey = encodePubkey(encodeSecp256k1Pubkey(accountFromSigner.pubkey));
+    let pubkey;
+    if (accountFromSigner.algo == "eth_secp256k1" || accountFromSigner.algo == "ethsecp256k1") {
```

Comment:
It would be good to add a comment here to explain why you need `eth_secp256k1` and `ethsecp256k1`.